neuralpredictors/layers/readouts/base.py

```diff
@@ -67,12 +67,21 @@ def resolve_reduction_method(self, reduction: Reduction = "mean", average: Optio
             reduction = "mean" if average else "sum"
         return reduction
 
-    def resolve_deprecated_gamma_readout(self, feature_reg_weight: float, gamma_readout: Optional[float]) -> float:
-        if gamma_readout is not None:
+    def resolve_deprecated_gamma_readout(
+        self, feature_reg_weight: Optional[float], gamma_readout: Optional[float], default: [float] = 1.0
+    ) -> float:
+        if feature_reg_weight is not None:
+            if gamma_readout is not None:
+                warnings.warn(
+                    "Use of 'gamma_readout' is deprecated. As 'feature_reg_weight' is defined, 'gamma_readout' is ignored."
+                )
+        elif gamma_readout is not None:
             warnings.warn(
                 "Use of 'gamma_readout' is deprecated. Please consider using the readout's feature-regularization parameter instead"
             )
             feature_reg_weight = gamma_readout
+        else:
+            feature_reg_weight = default
```

Comment:
I am checking if `gamma_readout is not None` only once because only one of 2 branches is going to be executed. I am not sure how you would like me to rewrite if that it would be checked once.

I can propose something like this but then  `feature_reg_weight ` would be check once independently in separate if-else branches but written twise, so I don't see any differences.

```
if gamma_readout is not None:
    if feature_reg_weight is not None:
        warnings.warn(
            "Use of 'gamma_readout' is deprecated. As 'feature_reg_weight' is defined, 'gamma_readout' is ignored."
            )
    else:
        feature_reg_weight = gamma_readout
        warnings.warn(
            "Use of 'gamma_readout' is deprecated. Please consider using the readout's feature-regularization parameter instead"
            )
else:
    feature_reg_weight = feature_reg_weight if feature_reg_weight else default
```

@cblessing24  Could you please share an example of what you would like to see instead?

Reply:
I see. I think the main part that tripped me up here was that there are two warnings that both warn about the same thing and I am having a hard time following the conditional logic.

```python
if gamma_readout is not None:
    warnings.warn("Use of 'gamma_readout' is deprecated. Use 'feature_reg_weight' instead.")

if feature_reg_weight is None:
    if gamma_readout is not None:
        feature_reg_weight = gamma_readout
    else:
        feature_reg_weight = default
```

Maybe something like this would be easier to understand?

Reply:
I wanted to keep the warning that gamma_readout value would be ignored if both are specified. I guess you think it's not necessary?

Reply:
You could still mention that in the warning.

Reply:
Thanks for the suggestion. I followed it with the updated warning